haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -101,6 +103,11 @@ instance IsChainParametersVersion cpv => Serialize (MintDistribution cpv) where
 instance IsChainParametersVersion cpv => HashableTo Hash.Hash (MintDistribution cpv) where
   getHash = Hash.hash . encode
 
+instance Arbitrary (MintDistribution 'ChainParametersV1) where
+  arbitrary = do
+    mps <- vector 2 `suchThat` (\[x, y] -> isJust $ addAmountFraction x y)
+    return $ MintDistribution MintPerSlotForCPV0None (mps !! 0) (mps !! 1)
```

Comment:
```suggestion
    (x, y) <- arbitrary `suchThat` (\(x, y) -> isJust $ addAmountFraction x y)
    return $ MintDistribution MintPerSlotForCPV0None x y
```

Reply:
I think it might be better to choose the first value from the entire range, and then the second from the rest, avoiding using `suchThat`.

Reply:
On consideration, probably this approach is fine given the fix to the PartsPerHundredThousands instance, since each choice of the two values has a 50% chance of being acceptable.